semcore/data-table/src/Body.tsx

```diff
@@ -58,12 +59,59 @@ class Body extends Component<AsProps, State> {
   firstRowRef = React.createRef<HTMLDivElement>();
   firstRowResizeObserver: ResizeObserver | null = null;
 
+  lockedCell: [HTMLElement | null, boolean] = [null, false];
+
   getRowHeight = () => {
     const { virtualScroll } = this.asProps;
     const rowHeightFromProps = typeof virtualScroll === 'object' && virtualScroll?.rowHeight;
     return rowHeightFromProps || this.state.rowHeight;
   };
 
+  handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.currentTarget === this.lockedCell[0]) {
+      const focusableChildren = Array.from(this.lockedCell[0].children).flatMap((node) =>
+        getFocusableIn(node as HTMLElement),
+      );
+
+      if (this.lockedCell[1]) {
+        if (e.key === 'Escape') {
+          this.lockedCell[0]?.focus();
+          this.lockedCell[1] = false;
+        }
+        if (e.key.startsWith('Arrow')) {
+          e.stopPropagation();
+        }
+        if (e.key === 'Tab') {
+          if (e.target === focusableChildren[0] && e.shiftKey) {
+            focusableChildren[focusableChildren.length - 1]?.focus();
+            e.preventDefault();
+          } else if (e.target === focusableChildren[focusableChildren.length - 1] && !e.shiftKey) {
+            focusableChildren[0]?.focus();
+            e.preventDefault();
+          }
+          e.stopPropagation();
+        }
+      } else if (e.key === 'Enter') {
+        this.lockedCell[1] = true;
+        focusableChildren[0]?.focus();
+      }
+    }
+  };
+
+  onFocusCell = (e: React.FocusEvent<HTMLElement, HTMLElement>) => {
+    if (e.target === e.currentTarget) {
```

Comment:
What the point of this check?

Reply:
that we focused on the cell, not on some element inside the cell